c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -3341,7 +3356,11 @@ class DeviceCachingAllocator {
         EventPool::Event event = std::move(e.first);
         Block* block = e.second;
 
-        cudaError_t err = C10_CUDA_ERROR_HANDLED(cudaEventQuery(*event));
+        cudaError_t err{};
+        {
+          at::cuda::CUDAStreamCaptureModeGuard g{cudaStreamCaptureModeRelaxed};
+          err = C10_CUDA_ERROR_HANDLED(cudaEventQuery(*event));
```

Comment:
Okay, this is wrong. During stream capture, your events are considered to be representing empty sets of work (they only have pending work upon calling cudaGraphLaunch(), but before the corresponding event record node), so cudaEventQuery will always return cudaSuccess. This doesn't make sense.

Reply:
Yes, you're right about this. The eventQuery will always return true because the job on the stream hasn’t actually executed before the graph is launched. I think I need to revisit this idea.